From f672950555967bef5d3ea58df2f28ae355758d53 Mon Sep 17 00:00:00 2001
From: Marshall Zhan <marshall.zhan.wiwynn@gmail.com>
Date: Tue, 20 Aug 2024 15:50:00 +0800
Subject: [PATCH] Fix the log file did not rotate after the system power off

use the end of the file as the file size.
---
 log-handler.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/log-handler.c b/log-handler.c
index d93b679..fbbc140 100644
--- a/log-handler.c
+++ b/log-handler.c
@@ -68,7 +68,9 @@ static int log_trim(struct log_handler *lh)
 		return -1;
 	}
 
-	lh->size = 0;
+	if (!rc) {
+		lh->size = 0;
+	}
 
 	return 0;
 }
@@ -135,7 +137,7 @@ static int log_create(struct log_handler *lh)
 		warn("Can't open log buffer file %s", lh->log_filename);
 		return -1;
 	}
-	pos = lseek(lh->fd, 0, SEEK_CUR);
+	pos = lseek(lh->fd, 0, SEEK_END);
 	if (pos < 0) {
 		warn("Can't query log position for file %s", lh->log_filename);
 		close(lh->fd);
-- 
2.25.1

